cmd/svcat/instance/provision_cmd.go

```diff
@@ -67,6 +68,8 @@ func NewProvisionCmd(cxt *command.Context) *cobra.Command {
 		RunE:    command.RunE(provisionCmd),
 	}
 	command.AddNamespaceFlags(cmd.Flags(), false)
+	cmd.Flags().StringVar(&provisionCmd.externalID, "external-id", "",
+		"The external ID of the instance for use with the OSB SB API (Optional)")
```

Comment:
This help text is not valid, OSBAPI does not know about external id. that is a construct of Service Catalog.

Reply:
The ExternalID of ServiceInstance is sent as InstanceID in OSB ProvisionRequest as can be seen below.
https://github.com/kubernetes-incubator/service-catalog/blob/2e1ca7f8cf2b5d45b0542602f0d12cb18bb05f98/pkg/controller/controller_instance.go#L1456

Also, the comment for the ExternalID field in ServiceInstance also says the same thing.
https://github.com/kubernetes-incubator/service-catalog/blob/2e1ca7f8cf2b5d45b0542602f0d12cb18bb05f98/pkg/apis/servicecatalog/types.go#L705-L708

Reply:
The ExternalID of ServiceBInding is sent as BindingID in OSB BindRequest.
https://github.com/kubernetes-incubator/service-catalog/blob/2e1ca7f8cf2b5d45b0542602f0d12cb18bb05f98/pkg/controller/controller_binding.go#L1038

Also, the comment for the ExternalID field of ServiceBinding says the same thing.
https://github.com/kubernetes-incubator/service-catalog/blob/2e1ca7f8cf2b5d45b0542602f0d12cb18bb05f98/pkg/apis/servicecatalog/types.go#L962-L965

Reply:
@amshuman-kr  the issue is that the OSB API doesn't use the term "external id".  That is a service catalog term.  In OSB API terms, the catalog external id maps to the ID field.  I think if you drop the word "external" from your help text it reads ok.

Reply:
Ah ok. Understood. I will make the changes.